bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md

```diff
@@ -0,0 +1,128 @@
+# STM32 Nucleo-144 BSP Introduction
+
+This document records the guide of the BSP (board support package) provided by the RT-Thread development team for the STM32 Nucleo-144 development boards.
+
+The document is covered in three parts:
+
+-  Resources Introduction
+- Quickly Get Started
+- Advanced Features
+
+By reading the ***Quickly Get Started*** section developers can quickly get their hands on this BSP and run RT-Thread on the board. More advanced features will be introduced in the Advanced Features section to help developers take advantage of RT-Thread to drive more on-board resources.
+
+
+
+## Resources Introduction
+
+[<img src="figures/stm32-nucleo-144.png" alt="board" style="zoom:50%;" />](figures/stm32-nucleo-144.jpg) 
+
+### Description 
+
+The STM32 Nucleo-144 board provides an affordable and flexible way for users to try out new concepts and build prototypes by choosing from the various combinations of performance and power consumption features, provided by the STM32 microcontroller. For the compatible boards, the internal or external SMPS significantly reduces power consumption in Run mode. The ST Zio connector, which extends the ARDUINO® Uno V3 connectivity, and the ST morpho headers provide an easy means of expanding the functionality of the Nucleo open development platform with a wide choice of specialized shields. The STM32 Nucleo-144 board does not require any separate probe as it integrates the ST-LINK debugger/programmer. The STM32 Nucleo-144 board comes with the STM32 comprehensive free software libraries and examples available with the STM32Cube MCU Package.
+
+### Features
+
+- Common features
+  - STM32 microcontroller in LQFP144 package 
+  - 3 user LEDs
+  - 2 user and reset push-buttons
+  - 32.768 kHz crystal oscillator
+  - Board connectors: 
+    -  SWD 
+    - ST Zio expansion connector including ARDUINO® Uno V3
+    - ST morpho expansion connector
+  - Flexible power-supply options: ST-LINK, USB VBUS or external sources
+  - On-board ST-LINK debugger/programmer with USB re-enumeration capability: mass storage, Virtual COM port, and debug port
+  - Comprehensive free software libraries and examples available with the STM32Cube MCU Package
+  - Support of a wide choice of Integrated Development Environments (IDEs) including IAR™, Keil®, and STM32CubeIDE
+- Board-specific features
+  - External or internal SMPS to generate Vcore logic supply
+  - Ethernet compliant with IEEE-802.3-2002
+  - USB OTG full speed or device only
+  - Board connectors: 
+    - USB with Micro-AB or USB Type-C™ 
+    - Ethernet RJ45
+  - Arm® Mbed Enabled™ compliant
+
+### **For more details about these boards, please refer to [ST Nucleo Official Website](https://www.st.com/en/evaluation-tools/stm32-nucleo-boards.html?querycriteria=productId=LN1847).**
```

Comment:
For more details about this board.

Reply:
Negative, "Nucleo-144" is a series of boards and has many different sub-types.

Reply:
Ohhh, I see then. Thanks for sharing.